### PR TITLE
Proxy fixes for CRIs

### PR DESCRIPTION
**Pull request by safihamid**
This will cover two CRIs reported in:

https://github.com/Azure/azure-webjobs-sdk-script/issues/2099
and
https://github.com/Azure/azure-webjobs-sdk-script/issues/2116

### CODE REVIEW

File: test/WebJobs.Script.Tests.Integration/ProxyEndToEndTests.cs [hunk 1]
```diff
@@ -3,52 +3,105 @@
...
+            string content = await response.Content.ReadAsStringAsync();
 
-            HttpResponseMessage response = (HttpResponseMessage)request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey];
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-            Assert.Equal(response.Headers.GetValues("myversion").ToArray()[0], "123");
+            Assert.True(content.StartsWith("Time:"));
+        }
+
+        [Theory]
+        [InlineData("test.txt")]
```
Comment (omkarmore83) [inline test/WebJobs.Script.Tests.Integration/ProxyEndToEndTests.cs]: Consider adding more extensions like html, js, css

Comment (christopheranderson) [reply test/WebJobs.Script.Tests.Integration/ProxyEndToEndTests.cs]: Would probably make sense to add a `.svc` extension, just for the sake of covering the CRI condition.

Comment (christopheranderson) [reply test/WebJobs.Script.Tests.Integration/ProxyEndToEndTests.cs]: nvm, I see it now

**Comment by christopheranderson** [comment]
:shipit:  - but should consider adding a test case for the "/" index case given that we broke that recently, too, right? Not sure if that's covered in this CRI, but it looked like it was missing in the e2e test coverage.

**Comment by safihamid** [comment]
@christopheranderson it’s the root check test here: https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/test/WebJobs.Script.Tests.E2E/ProxyEndToEndTests.cs

**Pull request merged**